core/templates/cowdata.h

```diff
@@ -146,8 +138,26 @@ class CowData {
 	void _unref();
 	void _ref(const CowData *p_from);
 	void _ref(const CowData &p_from);
-	USize _copy_on_write();
-	Error _realloc(Size p_alloc_size);
+
+	// Ensures that the backing buffer is at least p_size wide, and that
+	//  this CowData instance is the only reference to it. The buffer is
+	//  populated with as many element copies from the old array as possible.
+	// It is the responsibility of the caller to populate newly allocated
+	//  space up to p_size.
+	Error _fork_allocate(USize p_size);
+	Error _copy_on_write() { return _fork_allocate(size()); }
+
+	// Allocates a backing array of the given capacity. The reference count is
+	//  initialized to 1.
+	// It is the responsibility of the caller to populate the array and the new
+	//  size property.
+	Error _alloc(USize p_alloc_size);
+	// Re-allocates the backing array to the given capacity. The reference count is
+	//  initialized to 1.
+	// It is the responsibility of the caller to populate the array and the new
+	//  size property. The caller must also make sure there are no other references to
+	//  the data as pointers may be invalidated.
+	Error _realloc(USize p_alloc_size);
```

Comment:
Utilize comment width threshold of ~100
```suggestion
	// Ensures that the backing buffer is at least p_size wide, and that this CowData instance is
	// the only reference to it. The buffer is populated with as many element copies from the old
	// array as possible.
	// It is the responsibility of the caller to populate newly allocated space up to p_size.
	Error _fork_allocate(USize p_size);
	Error _copy_on_write() { return _fork_allocate(size()); }

	// Allocates a backing array of the given capacity. The reference count is initialized to 1.
	// It is the responsibility of the caller to populate the array and the new size property.
	Error _alloc(USize p_alloc_size);

	// Re-allocates the backing array to the given capacity. The reference count is initialized to 1.
	// It is the responsibility of the caller to populate the array and the new size property.
	// The caller must also make sure there are no other references to the data, as pointers may
	// be invalidated.
	Error _realloc(USize p_alloc_size);
```